### ISSUE TITLE
Always get an error

### ISSUE THREAD

**Issue by steverandy** [imagemin/imagemin-jpeg-recompress#9]
It seems that image has been optimized, but error is still thrown and process will exit.

```
events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: ssim at q=50 (40 - 60): 0.996457
ssim at q=55 (51 - 60): 0.996672
ssim at q=58 (56 - 60): 0.997332
ssim at q=59 (59 - 60): 0.997369
ssim at q=60 (60 - 60): 0.997396
Final optimized ssim at q=61: 0.997048
New size is 47% of original (saved 63 kb)
```

**Comment by arsgeografica** [comment]
Using 4.0.0 I am having this problem back again:

    events.js:72
            throw er; // Unhandled 'error' event
                  ^
    Error: ssim at q=62 (40 - 85): 1.000000
    ssim at q=50 (40 - 61): 0.999992
    ssim at q=44 (40 - 49): 0.999992
    ssim at q=41 (40 - 43): 0.999992
    ssim at q=40 (40 - 40): 1.000000
    Final optimized ssim at q=40: 1.000000
    New size is 174% of original (saved 0 kb)
    Output file is larger than input, aborting!

        at ChildProcess.<anonymous> (/run/media/cwygoda/Floh/mercury/node_modules/imagemin-jpeg-recompress/index.js:94:8)
        at ChildProcess.emit (events.js:98:17)
        at maybeClose (child_process.js:766:16)
        at Socket.<anonymous> (child_process.js:979:11)
        at Socket.emit (events.js:95:17)
        at Pipe.close (net.js:466:12)

**Comment by kevva** [comment]
Guess it's exits with an error code which results in an error https://github.com/imagemin/imagemin-jpeg-recompress/blob/master/index.js#L93-L96.

**Comment by shinnn** [comment]
jpeg-recompress always exits with code `1` regardless of the type of error.

I know it's too fragile, but we should test the error message with a regular expression on error.

```js
if (code) {
  if (/Output file is larger than input, aborting\!/.test(err)) {
    cb(null, file);
    return;
  }

  cb(new Error(err));
  return;
}
```

**Comment by shinnn** [comment]
Fixed in v4.1.0.

**Comment by arsgeografica** [comment]
Great job folks!
